src/Components/Aspire.Confluent.Kafka/AspireKafkaConsumerExtensions.cs

```diff
@@ -0,0 +1,175 @@
+                ? ConfluentKafkaCommon.ConsumerHealthCheckName
+                : string.Concat(ConfluentKafkaCommon.KeyedConsumerHealthCheckName, connectionName);
+
+            builder.TryAddHealthCheck(new HealthCheckRegistration(healthCheckName,
+                sp =>
+                {
+                    var connectionFactory = serviceKey is null
+                        ? sp.GetRequiredService<ConsumerConnectionFactory<TKey, TValue>>()
+                        : sp.GetRequiredKeyedService<ConsumerConnectionFactory<TKey, TValue>>(serviceKey);
+
+                    var options = new KafkaHealthCheckOptions();
+                    options.Configuration = new ProducerConfig(connectionFactory.Config.ToDictionary());
+                    options.Configuration.SocketTimeoutMs = 1000;
+                    options.Configuration.MessageTimeoutMs = 1000;
+                    options.Configuration.StatisticsIntervalMs = 0;
+                    return new KafkaHealthCheck(options);
+                },
+                failureStatus: default,
+                tags: default));
+        }
+    }
+
+    private static ConsumerConnectionFactory<TKey, TValue> CreateConsumerConnectionFactory<TKey, TValue>(IServiceProvider serviceProvider, Action<ConsumerBuilder<TKey, TValue>>? configureBuilder, KafkaConsumerSettings settings)
+        => new(CreateConsumerBuilder(serviceProvider, configureBuilder, settings), settings.Config!);
+    
+    private static ConsumerBuilder<TKey, TValue> CreateConsumerBuilder<TKey, TValue>(IServiceProvider serviceProvider, Action<ConsumerBuilder<TKey, TValue>>? configureBuilder, KafkaConsumerSettings settings)
+    {
+        settings.Validate();
+
+        ConsumerBuilder<TKey, TValue> builder = new(settings.Config);
+        ILogger logger = serviceProvider.GetRequiredService<ILoggerFactory>().CreateLogger(ConfluentKafkaCommon.LogCategoryName);
+        configureBuilder?.Invoke(builder);
+
+        try
+        {
+            void OnLog(IConsumer<TKey, TValue> _, LogMessage logMessage) =>
+                logger.Log((LogLevel)logMessage.LevelAs(LogLevelType.MicrosoftExtensionsLogging), 0, logMessage.Message, null, static (value, ex) => value);
+
+            builder.SetLogHandler(OnLog);
+        }
+        catch (InvalidOperationException)
+        {
+            logger.LogWarning("LogHandler is already set. Skipping... No logs will be written.");
+        }
+
+        if (settings.Metrics)
+        {
+            MetricsChannel channel = serviceProvider.GetRequiredService<MetricsChannel>();
+            void OnStatistics(IConsumer<TKey, TValue> _, string json)
+            {
+                if (string.IsNullOrEmpty(json))
+                {
+                    return;
+                }
+
+                // StatisticsHandler is called on the consume thread, so we need to offload the processing
+                // to avoid slowing the consumer down.
+                channel.Writer.TryWrite(json);
```

Comment:
If the Channel is full, we just drop the statistic? Is that the intention?

Reply:
Yes. The current size is 10_000. The size has been fixed to avoid a potential memory leak which may occur by creating an unbounded channel.